apple/internal/xcarchive.bzl

```diff
@@ -0,0 +1,90 @@
+"""
+Rule for packaging a bundle into a .xcarchive.
+"""
+
+load(
+    "@build_bazel_rules_apple//apple:providers.bzl",
+    "AppleBundleInfo",
+    "AppleDsymBundleInfo",
+)
+
+def _xcarchive_impl(ctx):
+    """
+    Implementation for xcarchive.
+
+    This rule uses the providers from the bundle target to re-package it into a .xcarchive.
+    The .xcarchive is a directory that contains the .app bundle, dSYM and other metadata.
+    """
+    bundle_info = ctx.attr.bundle[AppleBundleInfo]
+    dsym_info = ctx.attr.bundle[AppleDsymBundleInfo]
```

Comment:
Are dsyms produced automatically or does this still require manually setting some flags when invoking this rule? I don't see any transition to enforce that dsyms are produced when invoking this rule without any flag set.

Reply:
There's no transition here (yet) so I think it's currently expected to use the same flags as if you build the app target (i.e. `ios_application`).

Reply:
I see, we can add transitions later then.

Reply:
Yeah thats right, it'll be added with the extra outputs flag. But we should extend this to transition for dsyms automatically and some other things (like collect swift support).